erdpy/accounts.py

```diff
@@ -45,15 +45,15 @@ def get_all(self):
 
 
 class Account(IAccount):
-    def __init__(self, address: Any = None, pem_file: Union[str, None] = None, pem_index: int = 0, key_file: str = "", pass_file: str = "",
+    def __init__(self, address: Any = None, pem_file: Optional[str] = None, pem_index: int = 0, key_file: str = "", pass_file: str = "",
                  ledger: bool = False):
         self.address = Address(address)
         self.pem_file = pem_file
         self.pem_index = int(pem_index)
         self.nonce: int = 0
         self.ledger = ledger
 
-        if pem_file:
+        if self.pem_file:
```

Comment:
The type checker (`mypy`) was complaining, because the next line contains a call to `pem.parse()`.

The first argument of `pem.parse()` has the type `Union[str, Path]`, so it cannot accept a `None` value. On the other hand, the `pem_file` parameter has the type `Optional[str]`, so it's either `str` or `None`. Because there's an `if` before the call to `pem.parse()`, the type-checker knows that whatever we made the condition on cannot be `None`.

If we write `if self.pem_file` and then call `pem.parse(self.pem_file, ...)` then the type-checker knows that `self.pem_file` cannot be `None`, so the call is good. Alternatively, if we had `if pem_file` (like before) and then called `pem.parse(pem_file, ...)`, that would also have been good.

As long as we use the same variable in the `if` condition and in the call to `pem.parse()`, the type-checker will know that `pem.parse()` will never be called with `None`.

Previously, we had `if pem_file` and `pem.parse(self.pem_file, ...)`, so we had the condition on one variable, guaranteed by the `if` that it is not `None` at runtime, but the used the other variable (unverified) in the call to `pem.parse()` - which cannot take `None`.